modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -864,6 +864,15 @@ CV__DNN_INLINE_NS_BEGIN
      */
     CV_EXPORTS_W Net readNetFromONNX(const String &onnxFile);
 
+    /** @brief Reads a network model <a href="https://onnx.ai/">ONNX</a> from
+     *         in-memory buffer.
+     *  @param buffer memory address of the first byte of the buffer.
+     *  @param sizeBuffer size of the buffer.
+     *  @returns Network object that ready to do forward, throw an exception
+     *        in failure cases.
+     */
+    CV_EXPORTS_W Net readNetFromONNXBytes(const char* buffer, size_t sizeBuffer);
```

Comment:
This method cannot be wrapped because there is no way to bind `const char*` buffer in different languages. So please replace to `CV_EXPORTS`. And keep the origin name `readNetFromONNX`.

For bindings, please define
```cpp
CV_EXPORTS_W Net readNetFromONNX(const std::vector<uchar>& buffer);
```

See Caffe importer for reference.

Reply:
Thanks.  I've switched to using CV_EXPORTS, and renamed the function to drop the trailing "Bytes".

Unfortunately I cannot use std::vector<uchar> for my use case, since our ONNX model bytes are either embedded to the program binary, or memory-mapped in at run-time.  I could copy the bytes to std::vector<uchar>, but the point of this change is to avoid unnecessary memory-copy...

Reply:
@kohei-us, I mean you can define both methods with `const char*` and `std::vector<uchar>`. The second one might be used in Python and Java bindings.

Reply:
Or if you don't need Python and Java you can just keep `const char*` but with `CV_EXPORTS`, yes.

Reply:
@dkurt Ah, I got it.  I've modified it to add an overloaded variant as you suggested.